src/wirecraft/client/game.py

```diff
@@ -176,93 +176,138 @@ def drawmenu(self, device: Device) -> None:
-                        if device.rect.collidepoint(pygame.mouse.get_pos()):
-                            self.cables.append(
-                                Cable(
-                                    camera.world_to_screen(device.world_pos, self.resolution.size),
-                                    pygame.mouse.get_pos(),
-                                )
-                            )
-                            self.is_placing_cable = True
-                            break
-                else:
-                    for device in self.devices:
-                        if (
-                            device.rect.collidepoint(pygame.mouse.get_pos())
-                            and camera.world_to_screen(device.world_pos, self.resolution.size) != self.cables[-1].start
-                        ):
-                            self.is_placing_cable = False
-                            self.cables[-1].ended = True
-                            self.cables[-1].end = camera.world_to_screen(device.world_pos, self.resolution.size)
-                            break
-                self.dragging = True
-                self.last_mouse_pos = pygame.mouse.get_pos()
-                for device in self.devices:
-                    device.update_position(self.camera, self.resolution.size)
+            self.handle_mousebuttondown(event, camera)
 
         elif event.type == pygame.MOUSEBUTTONUP:
-            # Handle right click on devices
-            if event.button == MouseButtons.RIGHT.value and self.view == Gamestate.GAME and self.devices:
-                for device in self.devices:
-                    if device.rect.collidepoint(pygame.mouse.get_pos()):
-                        self.add_device_window(device)
-
-            if event.button == MouseButtons.LEFT.value:
-                self.dragging = False
-                self.last_mouse_pos = None
-                self.click_handled = False
-                # Handle close button
-                for window in self.windows:
-                    if (
-                        window.position[0] + window.size[0] - 40
-                        < pygame.mouse.get_pos()[0]
-                        < window.position[0] + window.size[0] - 15
-                        and window.position[1] + 10 < pygame.mouse.get_pos()[1] < window.position[1] + 35
-                    ):
-                        self.windows.remove(window)
-                # Handle click on buttons
-                for button in self.buttons:
-                    if (
-                        button.position[0] < pygame.mouse.get_pos()[0] < button.position[0] + button.size[0]
-                        and button.position[1] < pygame.mouse.get_pos()[1] < button.position[1] + button.size[1]
-                    ):
-                        button.action()
-
-        # Handle panning
+            self.handle_mousebuttonup(event)
+
         elif event.type == pygame.MOUSEMOTION and self.dragging and self.view == Gamestate.GAME:
```

Comment:
the additional conditions should maybe be included directly in the handle_mousemotion handler ?